packages/snjs/lib/services/features_service.ts

```diff
@@ -0,0 +1,178 @@
+} from './api/api_service';
+import { UuidString } from '@Lib/types';
+import { ContentType, Feature } from '@standardnotes/features';
+import { ItemManager } from './item_manager';
+import { UserFeaturesResponse } from './api/responses';
+import { SNComponentManager } from './component_manager';
+import { SNComponent } from '@Lib/models';
+import { SNWebSocketsService, WebSocketsServiceEvent } from './api/websockets_service';
+
+export class SNFeaturesService extends PureService<void> {
+  private roles: RoleName[] = [];
+  private removeApiServiceObserver?: () => void;
+  private removeWebSocketsServiceObserver?: () => void;
+
+  constructor(
+    private storageService: SNStorageService,
+    private apiService: SNApiService,
+    private itemManager: ItemManager,
+    private componentManager: SNComponentManager,
+    private webSocketsService: SNWebSocketsService,
+  ) {
+    super();
+
+    this.removeApiServiceObserver = apiService.addEventObserver(
+      async (eventName, data) => {
+        if (eventName === ApiServiceEvent.MetaReceived) {
+          const { userUuid, userRoles } = data as MetaReceivedData;
+          await this.updateRoles(
+            userUuid,
+            userRoles.map((role) => role.name)
+          );
+        }
+      }
+    );
+
+    this.removeWebSocketsServiceObserver = webSocketsService.addEventObserver(
+      async (eventName, data) => {
+        if (eventName === WebSocketsServiceEvent.UserRoleMessageReceived) {
+          const {
+            payload: { userUuid, currentRoles },
+          } = data as UserRolesChangedEvent;
+          await this.setRoles(currentRoles);
+          await this.updateFeatures(userUuid);
+        }
+      }
+    );
+  }
+
+  public async loadUserRoles(): Promise<void> {
+    this.roles =
+      (await this.storageService.getValue(StorageKey.UserRoles)) || [];
+  }
+
+  public async updateRoles(
+    userUuid: UuidString,
+    roles: RoleName[]
+  ): Promise<void> {
+    const userRolesChanged = this.haveRolesChanged(roles);
+    if (userRolesChanged) {
+      await this.setRoles(roles);
+      await this.updateFeatures(userUuid);
+    }
+  }
+
+  private async setRoles(roles: RoleName[]): Promise<void> {
+    this.roles = roles;
+    await this.storageService.setValue(StorageKey.UserRoles, this.roles);
+  }
+
+  private haveRolesChanged(roles: RoleName[]): boolean {
+    return (
+      roles.some((role) => !this.roles.includes(role)) ||
+      this.roles.some((role) => !roles.includes(role))
+    );
+  }
+
+  private async updateFeatures(userUuid: UuidString): Promise<void> {
+    const featuresResponse = await this.apiService.getUserFeatures(userUuid);
+    if (!featuresResponse.error && featuresResponse.data) {
+      await this.mapFeaturesToItems(
+        (featuresResponse as UserFeaturesResponse).data.features
+      );
+    }
+  }
+
+  private createItemDataForFeature(feature: Feature) {
+    return {
+      content_type: feature.contentType,
+      content: {
```

Comment:
You should use `FillItemContent` from SNJS here since there may be other required fields here.